From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Sid Manning <atx0x200@gmail.com>
Date: Fri, 3 Jul 2026 16:39:00 -0500
Subject: [PATCH] cli-kex: fix keytype/enum signkey_type mismatch

recv_msg_kexdh_reply() declares keytype as unsigned int but passes
its address to buf_get_pub_key(), which expects a
'enum signkey_type *'. This is caught as a hard error
(-Wincompatible-pointer-types) by clang/LLVM when building for
Hexagon, since enum signkey_type is not guaranteed to have the same
size/representation as unsigned int:

  src/cli-kex.c:130:44: error: incompatible pointer types passing
  'unsigned int *' to parameter of type 'enum signkey_type *'

Declare keytype with its actual type instead.

Signed-off-by: Sid Manning <atx0x200@gmail.com>
---
 src/cli-kex.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/cli-kex.c b/src/cli-kex.c
index 1111111..2222222 100644
--- a/src/cli-kex.c
+++ b/src/cli-kex.c
@@ -101,7 +101,8 @@
 void recv_msg_kexdh_reply() {

 	sign_key *hostkey = NULL;
-	unsigned int keytype, keybloblen;
+	enum signkey_type keytype;
+	unsigned int keybloblen;
 	unsigned char* keyblob = NULL;

 	TRACE(("enter recv_msg_kexdh_reply"))
--
2.34.1
